File: app/src/main/java/dev/yiray/qickmvivm/ui/search/SearchFragment.java
package dev.yiray.qickmvivm.ui.search;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import dev.yiray.qickmvivm.R;
import dev.yiray.qickmvivm.databinding.FragmentSearchBinding;

public class SearchFragment extends Fragment {

    private SearchViewModel searchViewModel;

    private FragmentSearchBinding binding;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        searchViewModel =
                new ViewModelProvider(this).get(SearchViewModel.class);

        binding = FragmentSearchBinding.inflate(inflater, container, false);
        searchViewModel.getUser().observe(getViewLifecycleOwner(), user -> {
            if (user != null) {
                String name = user.getName();

                if (name == null) {
                    name = "";
                }

                binding.information.setText(
                        String.format("%s, %s \n%s", name, user.getLogin(), user.getUrl()));
            } else {
                binding.information.setText(getString(R.string.hint_not_found));
            }
        });
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        binding.buttonSearch.setOnClickListener(v ->
                searchViewModel.findUser(binding.input.getText().toString()));
    }

}

File: app/src/main/java/dev/yiray/livedatamvi/ui/search/SearchFragment.kt
package dev.yiray.livedatamvi.ui.search

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import android.view.View
import dev.yiray.livedatamvi.R
import androidx.fragment.app.Fragment
import dev.yiray.livedatamvi.databinding.FragmentSearchBinding
import dev.yiray.livedatamvi.model.User

class SearchFragment : Fragment() {
    private var searchViewModel: SearchViewModel? = null
    private var binding: FragmentSearchBinding? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        searchViewModel = ViewModelProvider(this).get(SearchViewModel::class.java)
        binding = FragmentSearchBinding.inflate(inflater, container, false)

        searchViewModel!!.user.observe(viewLifecycleOwner, { user: User? ->
            if (user != null) {
                val name = user.name
                binding!!.information.text =
                    String.format("%s, %s \n%s", name, user.login, user.url)
            } else {
                binding!!.information.text = getString(R.string.hint_not_found)
            }
        })

        return binding!!.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding!!.buttonSearch.setOnClickListener { v: View? ->
            searchViewModel!!.findUser(
                binding!!.input.text.toString()
            )
        }
    }
}